### ISSUE TITLE
Undeprecate special handling of `${key}` syntax in item meta

### ISSUE THREAD

**Issue by erlehmann** [minetest/minetest#13966]
### Problem

Minetest supports setting both wield image and inventory image since @rubenwardy added it in this PR: https://github.com/minetest/minetest/pull/12614 

In the case of mods that generate bitmaps (e.g. maps, raycast scenes, text rendered with `unicode_text`) one might want to:

* store a bitmap in item meta so that:
   * the bitmap can later be modified easily
   * loading the item avoids dynamic media race conditions
   * generated wield images and inventory images work in world downloads
* refer to the stored bitmap as both the wield image and inventory image

Using `${key}` here to have both wield image and inventory image refer to the same thing is useful because:

* wield image and inventory image can be changed in one atomic operation
* only one bitmap needs to be stored in item meta if two or more are used

Here is some example code from my xcam mod that makes photos of in-game scenes using raycasting:
```
	meta:set_string(
		"xcam:texture",
		texture_string
	)
	meta:set_string(
		"inventory_image",
		"${xcam:texture}"
	)
	meta:set_string(
		"wield_image",
		"${xcam:texture}"
	)
```

This works currently, but quickly fills  the log with messages like this one:
```
2023-11-08 01:00:55: WARNING[Main]: Deprecated use of recursive resolution syntax in metadata: ${xcam:texture}
```

You can see the kind of item with a bitmap in meta that my `xcam` mod creates in this screenshot:

![minetest-mod-xcam-raytracer-3](https://github.com/minetest/minetest/assets/60905/6327a217-250b-4690-8dab-8abf52102b97)

### Solutions

Revisit the code and documentation added in this PR: https://github.com/minetest/minetest/pull/12970

As the feature works fine right now, the only change necessary would be the removal of the deprecation and runtime warning.

### Alternatives

* Do nothing. This means that items with inline bitmaps have to include two or three bitmaps. The main cost of this is that mods that set both `wield_item` and `inventory_item` to a dynamic string have to limit resolution and/or color depth of generated images to account for the worst case: A single 128×128 16bpp image with colored noise (i.e. incompressible) will not cause any harm in inventory, but two of them are probably going to result in “illegal” dropped item entities that have overlong meta.

* Add a texture modifier that does similar things, as @velartrill suggested in https://github.com/minetest/minetest/pull/12614#issuecomment-1248543340 – however, I doubt that this would be easy, as IMO this would make texture modifier cdoe a lot more context sensitive. I would prefer if that texture modifier processing would not get more convoluted.

### Additional context

I should have brought up that this feature is actually useful when https://github.com/minetest/minetest/issues/12577 was discussed. I somehow did not end up mentioning it.

I did bring this up in a comment on  https://github.com/minetest/minetest/pull/12614 – but that comment was marked as off-topic: https://github.com/minetest/minetest/pull/12614#issuecomment-1200497393 SImilarly, the reply from @velartrill that discussed it was also marked off-topic: https://github.com/minetest/minetest/pull/12614#issuecomment-1248543340 … I doubt that whoever marked it as offtopic did actually think that this would matter to users of the feature that was added in #12614, but IMO this issue shows that the interaction should have been considered.

**Comment by rubenwardy** [comment]
FYI: for craft items and tools, the wield image falls back to `inventory_image`. So you can just set `inventory_image` in this situation. That doesn't necessarily invalidate this request though

**Comment by erlehmann** [comment]
> FYI: for craft items and tools, the wield image falls back to `inventory_image`. So you can just set `inventory_image` in this situation. That doesn't necessarily invalidate this request though

This does help my situation indeed, thanks!

Of course, I can easily make up other cases where item meta affecting textures can be useful. In particular, an item that contains several inline textures (e.g. a very-low-res photo gallery or a custom magic book) has to hold one fewer texture and be slightly easier to code if the current picture could just be referenced, i.e. does not have to be copied.

Additionally, one might not want to use the exact image, but the inline image modified with some texture modifier (i.e. a photo or map frame). Do you have any idea if this actually works right now?

Basically, filenames are already references to data blobs, so I literally just want file-like access to textures stored in the item meta.

**Comment by sfan5** [comment]
The addition of `[png` increasingly sounds like a mistake, images should not be stored inline anywhere.

**Comment by hecktest** [comment]
> The addition of `[png` increasingly sounds like a mistake, images should not be stored inline anywhere.

They should not be **stored**, but that's just people not paying attention to what they're doing. [png was meant for: hud, forms, lowres raytraced skyboxes, _maybe_ uncomplicated entities where it's justified.

On the other hand, having to write a generated render buffer out to a file, give it a unique but ideally random filename, call an additional API, give it a table of options with a lambda callback, leave our current routine in an unfinished state, wait for said callback and _then_ run the code we originally wanted to, just to send a render buffer that we intend to discard, then wait until the user in question has disconnected to delete the generated trash, and also ensure that we clean it (and the in-progress callback state!) when the server crashes - that's the quality of life we need.

Nice of you to assume I even **have** a writeable filesystem post init. https://man.openbsd.org/unveil.2 https://man.openbsd.org/pledge.2

I was about to say that this might be a case where dynamic media is better, but no, OP has a point - embedding the stupid texture is very simple and avoids a lot of problems. This is an object-asset dependency problem that `[png` solves in a crappy way, and `dynamic_add` doesn't solve at all. (why should we send the contents of every photo to every player that connects? should we have to manually manage the memory for each player? how do we handle the inevitable race conditions? etc)

You could add another flag ("deferred") to dynamic add and scan each outgoing packet for references to said asset to know when to send it, but tomorrow someone else is gonna have another use case that defeats it.
